### PR TITLE
Turn cookbook recepies into examples

### PR DESCRIPTION
**Pull request by romanowski**

### CODE REVIEW

File: docs_checker/check.scala [hunk 1]
```diff
@@ -58,68 +60,105 @@ def parse(content: Seq[String], currentCommands: Seq[Commands], context: Context
...
+                lines.exists(_.contains(pattern)),
+                msg(s"Pattern: $pattern does not exisits in  any line in:\n$lastOutput")
+              )
+            }
+    }
+  finally if dest.isEmpty then ops.rm(ammPath(out))
+
+  // remove empty space at begining of all files
+  if dest.nonEmpty then
+    val header = s"File was generated from based on ${file}, do not edit manually!"
```
Comment (alexarchambault) [inline docs_checker/check.scala]: NIT
```suggestion
    val header = s"This file was generated from $file, do not edit manually!"
```

File: docs_checker/check.scala [hunk 2]
```diff
@@ -1,5 +1,6 @@
 // using scala 3.0.2
 // using "org.scalameta::munit:0.7.29"
+// using com.lihaoyi:ammonite-ops_2.13:2.4.0
```
Comment (alexarchambault) [inline docs_checker/check.scala]: Wouldn't [os-lib](https://github.com/com-lihaoyi/os-lib) (`com.lihaoyi::os-lib:0.7.5`) be enough? Ammonite-ops is more or less deprecated IIRC.

Comment (romanowski) [reply docs_checker/check.scala]: I just pick a random lib, I will change that

File: docs_checker/check.scala [hunk 3]
```diff
@@ -58,68 +59,118 @@ def parse(content: Seq[String], currentCommands: Seq[Commands], context: Context
...
         Files.list(path).iterator.asScala.filterNot(_.getFileName.toString.startsWith("."))
-      toCheck.toList.flatMap(checkPath)
+      toCheck.toList.flatMap(checkPath(dest))
   catch
     case e: Throwable =>
       e.printStackTrace()
       Seq(TestCase(path, Some(e)))
 
-def checkFile(file: Path) =
+def ammPath(p: Path) = os.Path(p.toAbsolutePath)
```
Comment (lwronski) [inline docs_checker/check.scala]: I think, this method can be removed if you migrate from `Java - Paths.*` to `os.lib`.

File: docs_checker/check.scala [hunk 4]
```diff
@@ -58,68 +59,118 @@ def parse(content: Seq[String], currentCommands: Seq[Commands], context: Context
...
+
+val fakeLineMarker = "//fakeline"
+
+def checkFile(file: Path, dest: Option[Path]) =
   val content  = Files.lines(file).iterator.asScala.toList
```
Comment (lwronski) [inline docs_checker/check.scala]:
```suggestion
  val content  = os.read.lines(file)
```
but, you have to replace `Path` to `os.Path`

**Pull request merged**